state/keys.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import "github.com/ava-labs/avalanchego/utils/set"
+
+const (
+	Read PermissionBit = 0
+	// TODO: Handle Allocate permission
+	Allocate      PermissionBit = 1
+	Write         PermissionBit = 2
+	PermissionLen PermissionBit = 8 // Sufficient to use a single byte
+)
+
+// StateKey holds the name of the key and its permission (Read/Allocate/Write). By default,
+// initialization of Keys with duplicate key will not work. And to prevent duplicate
+// insertions from overriding the original permissions, use the Add function below.
+// Permissions uses bits since transactions are expected to include this encoding
+type Keys map[string]Permissions
+
+// The access permission (Read/Allocate/Write) for each StateKey
+// Specifying RWrite is setting both the Read and Write bit
+type Permissions set.Bits
```

Comment:
I wonder if it is more efficient to just use a struct here like `{Allocate:bool, Write: bool, Read: bool}` instead of using this bitset.

This is not sent over the wire, so using a bitset seems a bit unnecessary. When it eventually is sent over the wire (when access pattern can't be inferred by the transaction), it makes sense.